accepted/future-releases/enhanced-enums/feature-specification.md

```diff
@@ -68,128 +66,175 @@ must redirect to a generative constructor._
 
 ## Semantics
 
-First we will add a `const Enum();` constructor to the `Enum` class, so that it can validly be a superclass of `enum` classes. _(The `Enum` class currently has a non-`const` default constructor that no-one can use because the class is abstract and cannot be extended, and in practice `enum` classes implement it instead. We now need to, at least pretend to, extend it so that mixins `on Enum` can be applied.)_
+We change the publicly visible API of the  `Enum` class to be declared as:
 
-The semantics of such an enum declaration is defined as being equivalent to *rewriting into a corresponding class declaration* as follows:
-
-- Declare a class with the same name and type parameters as the `enum` declaration.
-
-- Add `extends Enum`, where `Enum` refers to the type declared in `dart:core`.
+```dart
+class Enum {
+  // No default constructor (however we enforce that).
+  external int get index;
+  external String toString();
```

Comment:
With respect to the publicly visible API, `toString` may be omitted. We don't have the notion of a "primitive `toString` method", so clients should not have to know.

Maybe we could just say that "The `Enum` class behaves as if it were declared as [lines 72-76]", thus avoiding the direct reference to "publicly visible".

Reply:
ACK.
I'll retain `toString` just to suggest that it overrides `Object.toString`.